### PR TITLE
Change default font to core.text.DEFAULT_FONT

### PR DESCRIPTION
**Pull request by KeyWeeUsr**
Partial rebase of #2412 without the setting function. Closes #2412

### CODE REVIEW

**Comment by dessant** [comment]
It's a bit tricky, I also played with a fix but gave up at the time :P. See this thread for some of the pitfalls, and possible solutions: https://github.com/kivy/kivy/pull/2319.

The main benefit of setting a default font would be to not have to ship/load the font files provided by kivy, but if we must customize it by importing `kivy.core.text`, it's already too late because the module [loads](https://github.com/kivy/kivy/blob/e7299f4e7f6eb3ddf773c26a303af1dd52b90e27/kivy/core/text/__init__.py#L778) the default ones on import.

Setting/reading the default font from Config may be the only practical route at this time. That would involve declaring the font name and 4 paths, and finding a syntax which doesn't offend anyone. 😁

Here is one possible version:
```python
Config.set('kivy', 'default_font', '<Name>,<path_regular>,<path_italic>,<path_bold>,<path_bolditalic>')
```
A similar syntax is used already for some of the config options, but the comma could cause issues with certain paths.

**Comment by KeyWeeUsr** [comment]
Config seems to me like the was as even fullscreen is set this way. It's a _default_ value, it should be fetched from config, right?

Also, what's the problem with commas as separator? In the worst case we can do this:

    Config.set('kivy', 'default_font', '[<Name>,<path_regular>,<path_italic>,<path_bold>,<path_bolditalic>]')
 
and call Python's `ast` this way:

    ast.literal_eval('[<Name>,<path_regular>,<path_italic>,<path_bold>,<path_bolditalic>]')

which will return a list and is [quite safe](https://docs.python.org/3/library/ast.html#ast.literal_eval).

**Comment by KeyWeeUsr** [comment]
@dessant re the pitfalls, override it like this:

    Config.set(...)
    Config.write()

and the `config.ini` is overwritten, therefore on the **_next_** run of the app the font is properly changed.

**Comment by dessant** [comment]
I think this is now a decent way of solving it, if someone doesn't object until tomorrow, we can merge it.

File: kivy/config.py [hunk 1]
```diff
@@ -805,6 +811,14 @@ def name(self, value):
         elif version == 15:
             Config.setdefault('kivy', 'kivy_clock', 'default')
 
+        elif version == 16:
+            Config.setdefault('kivy', 'default_font', [
```
Comment (dessant) [inline kivy/config.py]: Should the value be a string here?

Comment (dessant) [inline kivy/config.py]: Sorry, we do it internally; https://github.com/kivy/kivy/blob/master/kivy/config.py#L433.

Comment (KeyWeeUsr) [inline kivy/config.py]: it's a list O.o

Comment (dessant) [inline kivy/config.py]: Yeah, i thought it would fail, but kivy handles it nicely.

**Comment by KeyWeeUsr** [comment]
However, there might be a problem with a need of **local** config:

    Config.read(<path to local .ini>)
    Config.set(...)
    Config.write()

Config somehow skips the local configuration and goes for `.kivy/config.ini`, but I think it's a bug in Config, not a problem of this PR.

File: kivy/config.py (hunk 1 above)
Comment (KeyWeeUsr) [inline kivy/config.py]: Yup and `literal_eval` comes to the rescue for converting it back 🍡 . It might be included in Config though.

**Comment by KeyWeeUsr** [comment]
Yep, was a config thing. Unfortunatelly, `kivy/__init__.py` sets it by default. It's nice though, because if prohibits dirty tricks ^^

```
import os
from os.path import dirname, abspath, join, exists
local_config = join(dirname(abspath(__file__)), 'local_kivy')
os.environ['KIVY_HOME'] = local_config

if not exists(os.environ['KIVY_HOME']):
    os.mkdir(local_config)

from kivy.config import Config
Config.set('kivy', 'default_font', [
    'Arial',
    local_config + '\\..\\arial.ttf',
    local_config + '\\..\\arial.ttf',
    local_config + '\\..\\arial.ttf',
    local_config + '\\..\\arial.ttf',
])
Config.write()

from kivy.lang import Builder
from kivy.base import runTouchApp
runTouchApp(Builder.load_string('''
Button:
    text: 'blaaaaaaaaaaaaa'
'''))
```

First run is still "cached", because Config isn't reloaded, but even after `import kivy` and some other stuff I tried, it was still the default font until the app was killed and ran for the second time, when the local config was finally read.

Any ideas how to restart Config after `Config.write()`?

ref #4826

**Comment by dessant** [comment]
@KeyWeeUsr, super, it works well with my simple test.
```python
from kivy.config import Config
Config.set('kivy', 'default_font', [
    'AveriaSansLibre',
    'AveriaSansLibre-Regular.ttf',
    'AveriaSansLibre-Italic.ttf',
    'AveriaSansLibre-Bold.ttf',
    'AveriaSansLibre-BoldItalic.ttf',
])

from kivy.app import App
from kivy.lang import Builder


kv = """
Label:
    font_size: dp(30)
    text: "I feel pretty,\\nOh, so pretty,"
"""


class TestApp(App):
    def build(self):
        return Builder.load_string(kv)

if __name__ == '__main__':
    TestApp().run()
```

Il do a small change, mainly for allowing to skip italic, bold and bolditalic.

If someone has an idea for a different config value syntax, please share.

**Pull request merged**